src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,240 @@ class RigidBodyComponentSystem extends ComponentSystem {
         return results;
     }
 
+    /**
+     * Perform a collision check on the world and return all entities the shape hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {object} shape - The shape to use for collision.
+     * @param {number} shape.axis - The local space axis with which the capsule, cylinder or cone shape's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} shape.halfExtends - The half-extents of the box in the x, y and z axes.
```

Comment:
The engine tends to use `halfExtents` instead of `halfExtends`. Search the source to see what I mean.

Reply:
My bad. Fixed in last commit!